designer/server/src/main/scala/pl/touk/nussknacker/ui/server/AkkaHttpBasedRouteProvider.scala

```diff
@@ -855,6 +878,27 @@ class AkkaHttpBasedRouteProvider(
     additionalUIConfigProviderFactory.create(config, sttpBackend)
   }
 
+  private def createCustomHttpServiceProvider(
+      config: Config,
+      processService: ProcessService
+  ): Resource[IO, CustomHttpServiceProvider] = {
+    Multiplicity(
+      ScalaServiceLoader.load[CustomHttpServiceProviderFactory](getClass.getClassLoader)
+    ) match {
+      case Empty() =>
+        Resource.pure[IO, CustomHttpServiceProvider](CustomHttpServiceProvider.noop)
+      case One(providerFactory) =>
+        providerFactory.create(
+          config,
+          NussknackerServicesForCustomHttpService(new ScenarioServiceImpl(processService)),
+        )
+      case Many(moreThanOne) =>
+        throw new IllegalArgumentException(
+          s"More than one CustomHttpServiceProviderFactory instance found: $moreThanOne"
```

Comment:
it's bad? we cannot load more than one? IMO we could but each provider should have a name that will be included in the endpoint path

Reply:
Ok, added possibility to have multiple providers. With name and detecting name conflicts.